hs_core/hydroshare/utils.py

```diff
@@ -754,6 +774,20 @@ def resource_file_add_process(resource, files, user, extract_metadata=False,
     return resource_file_objects
 
 
+def create_empty_contents_directory(resource):
+    if resource:
```

Comment:
Since this function is internally used from another function probably, I don't see the need for this check. Just a thought.